engine/class_modules/paladin/sc_paladin_protection.cpp

```diff
@@ -113,6 +118,10 @@ struct avengers_shield_t : public avengers_shield_base_t
       p() -> cooldowns.avengers_shield -> reset( false );
       p() -> buffs.moment_of_glory -> decrement( 1 );
     }
+    if ( p() -> legendary.holy_avengers_engraved_sigil -> ok() && rng().roll( holy_avenger_proc_chance ) )
+    {
+      p() -> cooldowns.avengers_shield -> reset( false );
+    }
```

Comment:
You may want to introduce a proc object similar paladin_t::procs.grand_crusader to track the AS resets from this legendary effect.